From e40381c06423a95c3cb3e89bff39d7777fc2c3b0 Mon Sep 17 00:00:00 2001
From: Piotr Zalewski <pZ010001011111@proton.me>
Date: Thu, 5 Dec 2024 19:06:13 +0100
Subject: [PATCH 4/4] rockchip/drm: vop2: mv color mgmt changed check out of
 set gamma LUT

Currently check if color management changed happens always inside
try_set_gamma_lut. After coming out of suspend, during the modeset, it
prevents gamma LUT from being rewritten in atomic_enable.
---
 drivers/gpu/drm/rockchip/rockchip_drm_vop2.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/rockchip/rockchip_drm_vop2.c b/drivers/gpu/drm/rockchip/rockchip_drm_vop2.c
index 58439ee8a52c..cd8311a7132a 100644
--- a/drivers/gpu/drm/rockchip/rockchip_drm_vop2.c
+++ b/drivers/gpu/drm/rockchip/rockchip_drm_vop2.c
@@ -1601,7 +1601,7 @@ static void vop2_crtc_atomic_try_set_gamma(struct vop2 *vop2,
 					struct drm_crtc_state *crtc_state)
 {
 
-	if (!vop2->lut_regs || !crtc_state->color_mgmt_changed)
+	if (!vop2->lut_regs)
 		return;
 
 	if (!crtc_state->gamma_lut) {
@@ -2671,7 +2671,7 @@ static void vop2_crtc_atomic_flush(struct drm_crtc *crtc,
 
 	// NOTE: in case of modeset gamma lut update
 	// already happened in atomic enable
-	if (!drm_atomic_crtc_needs_modeset(crtc_state))
+	if (!drm_atomic_crtc_needs_modeset(crtc_state) && crtc_state->color_mgmt_changed)
 		vop2_crtc_atomic_try_set_gamma_locked(vop2, vp, crtc, crtc_state);
 
 	vop2_post_config(crtc);
-- 
2.48.0

